rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java

```diff
@@ -19,164 +19,332 @@
+            blockNumberFederationActivationFingerroot = newFederationCreationBlockNumber + newFederationActivationAgeFingerroot;
+
+            // create old and new federations
+            P2shErpFederationBuilder p2shErpFederationBuilder = new P2shErpFederationBuilder();
+            oldFederation = p2shErpFederationBuilder
+                .withCreationBlockNumber(oldFederationCreationBlockNumber)
+                .build();
+            List<BtcECKey> newFederationKeys = BitcoinTestUtils.getBtcEcKeysFromSeeds(
+                new String[]{"fa01", "fa02", "fa03", "fa04", "fa05", "fa06", "fa07", "fa08", "fa09"}, true
+            );
+            newFederation = p2shErpFederationBuilder
+                .withMembersBtcPublicKeys(newFederationKeys)
+                .withCreationBlockNumber(newFederationCreationBlockNumber)
+                .build();
+
+            // save federations in storage
+            StorageAccessor storageAccessor = new InMemoryStorage();
+            storageProvider = new FederationStorageProviderImpl(storageAccessor);
+            storageProvider.setOldFederation(oldFederation);
+            storageProvider.setNewFederation(newFederation);
+        }
 
-        // Act
-        Federation activeFederation = federationSupport.getActiveFederation();
+        @ParameterizedTest
+        @MethodSource("expectedFederation_args")
+        void getActiveFederation_returnsExpectedFederationAccordingToActivationAgeAndActivations(long currentBlock, Federation expectedFederation,
+                                                                                                               ActivationConfig.ForBlock activations) {
+            Block executionBlock = mock(Block.class);
+            when(executionBlock.getNumber()).thenReturn(currentBlock);
+
+            federationSupport = federationSupportBuilder
+                .withFederationConstants(federationMainnetConstants)
+                .withFederationStorageProvider(storageProvider)
+                .withRskExecutionBlock(executionBlock)
+                .withActivations(activations)
+                .build();
+
+            Federation activeFederation = federationSupport.getActiveFederation();
+            assertThat(activeFederation, is(expectedFederation));
+        }
+        private Stream<Arguments> expectedFederation_args() {
+            return Stream.of(
+                Arguments.of(blockNumberFederationActivationHop - 1, oldFederation, hopActivations), // new federation shouldn't be active
+                Arguments.of(blockNumberFederationActivationHop + 1, newFederation, hopActivations), // new federation should be active
+                Arguments.of(blockNumberFederationActivationFingerroot - 1, oldFederation, fingerrootActivations), // new federation shouldn't be active
+                Arguments.of(blockNumberFederationActivationFingerroot + 1, newFederation, fingerrootActivations) // new federation should be active
```

Comment:
Maybe check using the exact activation block number? And not +1, to be sure we cover that corner case

Reply:
done